src/java.base/share/classes/java/text/ListFormat.java

```diff
@@ -0,0 +1,563 @@
+     * @param input The list of input strings to format. There should at least
+     *              one String element in this list, otherwise an {@code IllegalArgumentException}
+     *              is thrown.
+     * @throws IllegalArgumentException if the length of {@code input} is zero.
+     * @throws NullPointerException if {@code input} is null.
+     */
+    public String format(List<String> input) {
+        Objects.requireNonNull(input);
+        return format(input, new StringBuffer(),
+                DontCareFieldPosition.INSTANCE).toString();
+    }
+
+    /**
+     * Formats an object and appends the resulting text to a given string
+     * buffer. The object should either be a List or an array of Objects.
+     * If the {@code pos} argument identifies a field used by the format,
+     * then its indices are set to the beginning and end of the first such
+     * field encountered.
+     *
+     * @param obj    The object to format. Must be a List or an array
+     *               of Object.
+     * @param toAppendTo    where the text is to be appended
+     * @param pos    A {@code FieldPosition} identifying a field
+     *               in the formatted text
+     * @return       the string buffer passed in as {@code toAppendTo},
+     *               with formatted text appended
+     * @throws    NullPointerException if {@code toAppendTo} or
+     *            {@code pos} is null
+     * @throws    IllegalArgumentException if it cannot format the given
+     *            object
+     */
+    @Override
+    public StringBuffer format(Object obj, StringBuffer toAppendTo, FieldPosition pos) {
+        Objects.requireNonNull(toAppendTo);
+        Objects.requireNonNull(pos);
+        if (obj instanceof Object[] objs) {
+            return generateMessageFormat(objs).format(objs, toAppendTo, pos);
+        } else if (obj instanceof List<?> objs) {
+            var a = objs.toArray(new Object[0]);
+            return generateMessageFormat(a).format(a, toAppendTo, pos);
+        } else {
+            throw new IllegalArgumentException("The object to format should be an Object list");
+        }
+    }
+
+    /**
+     * {@return the parsed list of Strings from the {@code source} String}
+     *
+     * Note that {@link #format(List)} and this method
+     * may not guarantee a round-trip, if the input strings contain ambiguous
+     * delimiters. For example, a String array {@code ["a, b,", "c"]} will be
+     * formatted as {@code "a, b, and c"}, but may be parsed as
+     * {@code ["a", "b", "c"]}.
+     *
+     * @param source the String to parse, not null.
+     * @throws ParseException if parse failed
+     * @throws NullPointerException if source is null
+     */
+    public List<String> parse(String source) throws ParseException {
+        var pp = new ParsePosition(0);
+        if (parseObject(source, pp) instanceof List<?> orig) {
+            return orig.stream().map(Object::toString).toList();
```

Comment:
In what cases can it return something other than String?
The map to Object::String should not be necessary if parseObject returns Strings consistently.

Reply:
Yes, a simple cast to String should suffice.